File: contracts/rust/helloworld/helloworld.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package helloworld

import "github.com/iotaledger/wasp/packages/vm/wasmlib"

func funcHelloWorld(ctx *wasmlib.ScFuncContext, params *FuncHelloWorldParams) {
	ctx.Log("Hello, world!")
}

func viewGetHelloWorld(ctx *wasmlib.ScViewContext, params *ViewGetHelloWorldParams) {
	ctx.Log("Get Hello world!")
	ctx.Results().GetString(VarHelloWorld).SetValue("Hello, world!")
}

File: contracts/rust/helloworld/src/helloworld.rs
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use wasmlib::*;

use crate::*;

pub fn func_hello_world(ctx: &ScFuncContext, _f: &HelloWorldContext) {
    ctx.log("Hello, world!");
}

pub fn view_get_hello_world(_ctx: &ScViewContext, f: &GetHelloWorldContext) {
    f.results.hello_world().set_value("Hello, world!");
}
